showkase-processor/src/main/java/com/airbnb/android/showkase/processor/writer/ShowkaseMethodsWriter.kt

```diff
@@ -0,0 +1,154 @@
+            CONTEXT_PARAMETER_NAME, CONTEXT_CLASS_NAME
+        )
+        returns(INTENT_CLASS_NAME)
+        addCode(
+            CodeBlock.Builder()
+                .indent()
+                .addStatement(
+                    "val intent = %T(%L, %T::class.java)",
+                    INTENT_CLASS_NAME,
+                    CONTEXT_PARAMETER_NAME,
+                    SHOWKASE_BROWSER_ACTIVITY_CLASS_NAME
+                )
+                .addStatement(
+                    "intent.putExtra(%S, %S)",
+                    SHOWKASE_ROOT_MODULE_KEY,
+                    "$rootModulePackageName.$rootModuleClassName"
+                )
+                .addStatement(
+                    "return intent"
+                )
+                .unindent()
+                .build()
+        )
+        addOriginatingElement(rootElement)
+    }
+        .build()
+
+    private fun generateMetadataFunction(
+        rootElement: Element,
+        classKey: String
+    ) = FunSpec.builder(METADATA_FUNCTION_NAME).apply {
+        returns(SHOWKASE_ELEMENTS_METADATA_CLASS_NAME)
+        addCode(
+            CodeBlock.Builder()
+                .indent()
+                .addStatement("try {")
+                .indent()
+                .addStatement(
+                    "val showkaseComponentProvider = Class.forName(\"${classKey}Codegen\").newInstance() as %T",
+                    SHOWKASE_PROVIDER_CLASS_NAME
+                )
+                .generateShowkaseElementProperty(
+                    "componentsMap", 
+                    "getShowkaseComponents", 
+                    "group"
+                )
+                .generateShowkaseElementProperty(
+                    "colorsMap", 
+                    "getShowkaseColors", 
+                    "colorGroup"
+                )
+                .generateShowkaseElementProperty(
+                    "typographyMap", 
+                    "getShowkaseTypography", 
+                    "typographyGroup"
+                )
+                .addStatement(
+                    "return %T(%L, %L, %L)",
+                    SHOWKASE_ELEMENTS_METADATA_CLASS_NAME,
+                    "componentsMap",
+                    "colorsMap",
+                    "typographyMap"
+                )
+                .unindent()
+                .addStatement("} catch(exception: ClassNotFoundException) {")
+                .indent()
+                .addStatement("return %T()", SHOWKASE_ELEMENTS_METADATA_CLASS_NAME)
+                .unindent()
+                .addStatement("}")
+                .unindent()
+                .build()
+        )
+        addOriginatingElement(rootElement)
+    }
+        .build()
+
+    private fun CodeBlock.Builder.generateShowkaseElementProperty(
+        variableName: String,
+        elementMethodName: String,
+        groupPropertyName: String
+    ) = addStatement(
+        "val $variableName = %L.$elementMethodName().groupBy { it.$groupPropertyName }",
```

Comment:
I'm actually going to update this and simply return a `Set` instead of a map that has the group name as key. I think returning the flat list should be good enough and I should not imply my preference by grouping this as it can be inferred from the metadata anyway. What do y'all think?